src/main/java/user_interface/login_leaderboard/FileWriter.java

```diff
@@ -1,13 +1,127 @@
  * When a new score is set by user, file writer will update the score.
  */
-public class FileWriter {
-    public void updateNewUser(User newUser){}
+public class FileWriter implements IFileOutput {
+    private final String filePath = "C:\\Users\\arifa\\IdeaProjects\\course-project-group-93\\" +
+            "src\\main\\java\\user_interface\\login_leaderboard\\data.csv";
+
+    @Override
+    public String getUsername() {
+        return "" ;
+    }
+
+    /**
+     * Update data.csv to include a new row with the given information.
+     * @param name : username
+     * @param password : user password
+     * @param email : user email
+     */
+    @Override
+    public void updateNewUser(String name, String password, String email){
+        try {
+            BufferedWriter output;
+            output = new BufferedWriter(new java.io.FileWriter(filePath, true));
+            PrintWriter print = new PrintWriter(output);
+
+            String text = name + "," + password + "," + email + "," + "0" + ","
+                    + "0" + "," + "0";
+
+            print.println(text);
+
+            output.close();
+            print.close();
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Replace the users score in the CSV file if the current stamina is greator
+     * than the given one.
+     *
+     * @param score : Player stamina
+     * @param level : Easy, medium, or hard
+     * @param username : The current user's username
+     * @throws IOException : Data csv file
+     */
+    @Override
+    public void updateScore(int score, String level, String username) throws IOException {
+        File tmp = File.createTempFile("tmp", "");
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(filePath));
+
+            BufferedWriter output;
+            output = new BufferedWriter(new java.io.FileWriter(tmp, true));
+            PrintWriter print = new PrintWriter(output);
+
+            print.println(reader.readLine()); //header
+            String currLine;
+            String newLine = "";
+
+            while((currLine = reader.readLine()) != null) {
+                String[] values = currLine.split(",");
+                if (values[0].equals(username)){
+                    print.println(updateScoreHelper(score, level, newLine, values));
+                }
+                else {
+                    print.println(currLine);
+                }
+            }
+
+            print.close();
+            output.close();
+            reader.close();
+
+            File old = new File(filePath);
+            if (old.delete())
+                tmp.renameTo(old);
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
 
+    /**
+     * Helper method for updating score.
+     */
+    public String updateScoreHelper(int score, String level, String newLine, String[] values){
```

Comment:
Since it's a helper method it could be private? Also if you could add @params that would be great!

Reply:
Good idea!